template.hbs

```diff
@@ -309,85 +344,73 @@
+          <th class="table-headers" colspan="2">{{@key}} {{this.Key}}</th>
           <td class="table-cells">{{this.Value}} {{this.Unit}}</td>
           <td class="table-cells">{{this.Interpretation}}</td>
         </tr>
         {{/each}}
       </table>
       {{/ifEqual}}
-
-      {{!-- Inspection-> ChemicalComposition --}}
-      {{#ifEqual @key "ChemicalComposition"}}
-      {{#each this}}
-      {{#typeOf this 'object'}}
-      {{!-- ChemicalComposition-> SupplementaryInformation --}}
-      {{#ifEqual @key "SupplementaryInformation"}}
-      <h3 class="third-level-title second-level" style="float: left;">
-        {{ i18n @key "otherFields" ../../Certificate.CertificateLanguages }}</h3>
-      <table class="main-table second-level">
-        {{#each this}}
+     {{#ifEqual @key "ChemicalComposition"}}
+     <table class="main-table" style="margin-top: 0px;">
         <tr class="table-rows">
-          <th class="table-headers">{{@key}} {{this.Key}}</th>
-          <td class="table-cells">{{this.Value}} {{this.Unit}}</td>
-          <td class="table-cells">{{this.Interpretation}}</td>
+          <th></th>
+          {{#each this}}
+          <th class="table-headers">{{@key}}</th>
+          {{/each}}
         </tr>
-        {{/each}}
-      </table>
-      {{!-- ChemicalComposition-> ChemicalElement --}}
-      {{else}}
-      <table class="main-table" style="margin-top: 0px;">
         <tr class="table-rows">
-          {{!-- No translation ? --}}
-          <th class="table-headers">{{@key}}</th>
+          <th>Symbol</th>
           {{#each this}}
-          <td class="caption table-cells">{{@key}}</td>
+          <th class="table-headers">{{this.Symbol}}</th>
           {{/each}}
         </tr>
         <tr class="table-rows">
-          <td class="caption table-cells"></td>
+          <th>Actual [%]</th>
           {{#each this}}
-          <td>{{this}}</td>
+          <th class="table-headers">{{this.Actual}}</th>
           {{/each}}
         </tr>
       </table>
-      {{/ifEqual}}
-      {{!-- ChemicalComposition-> C70 --}}
-      {{else}}
-      <table class="main-table" style="margin-top: 0px;">
+      {{@key.SupplementaryInformation}}
+      {{#if @key.SupplementaryInformation}}
+      <h3 class="third-level-title">
+        {{ i18n @key "otherFields" ../../Certificate.CertificateLanguages }}</h3>
+      <table class="main-table">
+        {{#each @key.SupplementaryInformation}}
         <tr class="table-rows">
-          <th class="table-headers">{{@key}}
-            {{ i18n @key "certificateFields" ../../Certificate.CertificateLanguages }}</th>
-          <td class="table-cells">{{this}}<em class="caption">(Section)</em></td>
+          <th class="table-headers" colspan="2">{{@key}} {{this.Key}}</th>
+          <td class="table-cells">{{this.Value}} {{this.Unit}}</td>
+          <td class="table-cells">{{this.Interpretation}}</td>
         </tr>
+        {{/each}}
       </table>
```

Comment:
I have some doubts regarding that block, but if you tested it and it works then forget about it.
For sure, the rendering   `{{@key.SupplementaryInformation}}` should be done after the if.

Reply:
It doesn't work as expected 🤦🏻‍♂️.
1. Regex in schema wrong - as all elements up to C109 are reserved for chemical elements the idea was to have C110 to C120 for Supplementary Information.
2. And you are right the code doesn't work (handlebarsjs becomes slightly confusing at a certain point).
Can you please take care of that matters?

Reply:
Fixed locally